lib/Sema/SemaBounds.cpp

```diff
@@ -936,6 +934,20 @@ namespace {
         return true;
       }
 
+      if (CK == CK_DynamicPtrBounds || CK == CK_AssumePtrBounds) {
```

Comment:
Sorry, I just noticed this.   We should only be requiring bounds for dynamic_bounds_casts.  We should not require bounds for assume ptr bounds.

assume_bound_cast is meant to be used for things like code that has to cast a hardware address (just an integer) to a checked pointer type:
```
// memory-mapped hardware buffer that is 128 bytes in size begins at 0x4000
array_ptr<char> buf : count(128) = assume_bounds_cast<array_ptr<char>>(0x4000, 128);
```

We will want to require that the subexpression for both dynamic_bound_cast and assume_bounds_cast opertors have a type suitable for casting to a pointer. It should have a pointer or an integeral type.  I will open a separate work item for that.   I don't want that check added to this pull request.  I checked the specification and noticed that it was silent on the type of the first operand.   I will file a work item for that too.